examples/server.rs

```diff
@@ -1,5 +1,6 @@
 use futures::future;
 use jsonrpc_core::{BoxFuture, Result};
+use lsp_types::request::GotoImplementationResponse;
```

Comment:
I think that this `use` statement should be combined with the existing `tower_lsp::lsp_types::request` import on line 5, similar to how it's done [here](https://github.com/ebkalderon/tower-lsp/blob/e2db3532d5d7b8dc38585e77d787f3743a49b845/src/lib.rs#L11).

End-users shouldn't have to depend on `lsp-types` directly in order to interact with this library, and can import from the re-exported symbols through `tower-lsp` in order to keep versions consistent.